beacon/blockchain/payload.go

```diff
@@ -270,11 +266,7 @@ func (s *Service) optimisticPayloadBuild(
 	payload := blk.GetBody().GetExecutionPayload()
 	if _, _, err = s.localBuilder.RequestPayloadAsync(
 		ctx, st,
-		slot,
 		nextPayloadTimestamp,
-		// The previous block root is simply the root of the block we just
-		// processed.
-		blk.HashTreeRoot(),
```

Comment:
In case you are wondering, we don't need to set latest header in state here because we call `ProcessSlots` which does it already